src/main/java/mekanism/common/content/gear/mekatool/ModuleBlastingUnit.java

```diff
@@ -0,0 +1,111 @@
+import mekanism.api.gear.ICustomModule;
+import mekanism.api.gear.IModule;
+import mekanism.api.gear.config.IModuleConfigItem;
+import mekanism.api.gear.config.ModuleConfigItemCreator;
+import mekanism.api.gear.config.ModuleEnumData;
+import mekanism.api.text.EnumColor;
+import mekanism.api.text.IHasTextComponent;
+import mekanism.common.MekanismLang;
+import mekanism.common.block.BlockBounding;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+@ParametersAreNonnullByDefault
+public class ModuleBlastingUnit implements ICustomModule<ModuleBlastingUnit> {
+
+    private IModuleConfigItem<BlastRadius> blastRadius;
+
+    @Override
+    public void init(IModule<ModuleBlastingUnit> module, ModuleConfigItemCreator configItemCreator) {
+        blastRadius = configItemCreator.createConfigItem("blast_radius", MekanismLang.MODULE_BLASTING_RADIUS,
+                new ModuleEnumData<>(BlastRadius.class, module.getInstalledCount() + 1, BlastRadius.LOW));
+    }
+
+    public int getBlastRadius() {
+        return blastRadius.get().getRadius();
+    }
+
+    public static Map<BlockPos, BlockState> findPositions(Level world, BlockPos targetPos, BlockState targetState, Direction axis, int radius) {
+        int minX = 0, minY = 0, minZ = 0;
+        int maxX = 0, maxY = 0, maxZ = 0;
+        switch (axis) {
+            case UP, DOWN -> {
+                minX = -radius;
+                maxX = radius;
+                minY = maxY = 0;
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case EAST, WEST -> {
+                minX = maxX = 0;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = -radius;
+                maxZ = radius;
+            }
+            case NORTH, SOUTH -> {
+                minX = -radius;
+                maxX = radius;
+                minY = (radius == 0) ? 0 : -1;
+                maxY = (radius == 0) ? 0 : (2 * radius - 1);
+                minZ = maxZ = 0;
+            }
+        }
+        Map<BlockPos, BlockState> found = new HashMap<>();
+        for (BlockPos nextPos : BlockPos.betweenClosed(targetPos.offset(minX, minY, minZ), targetPos.offset(maxX, maxY, maxZ))) {
+            BlockState nextState = world.getBlockState(nextPos);
+            if (canBlastBlock(world, nextPos, nextState)) {
+                found.put(nextPos.immutable(), nextState);
+            }
+        }
+        return found;
+    }
+
+    public static boolean canBlastBlock(Level world, BlockPos pos, BlockState state) {
+        return !state.isAir() && !state.getMaterial().isLiquid() && !(state.getBlock() instanceof BlockBounding) && state.getDestroySpeed(world, pos) > 0;
```

Comment:
I wonder if it would make more sense from a mechanic standpoint to not allow blasting fluid logged blocks in general. I can see it both ways

Reply:
I more meant from a balance perspective. Similar to the fact that creepers/explosions when in water don't damage blocks.

Reply:
I'm leaning towards leaving it, vein mining also mines them and I think it makes sense for it to be consistent.